protocols/kad/src/protocol.rs

```diff
@@ -188,32 +227,9 @@ where
     type Error = io::Error;
```

Comment:
Just seeing that we actually never return this error! Meaning we can probably simplify a few things here by replacing this with `void::Void`. Should be done in a separate PR though to keep the scope small. Let me know if you are interested :)

Reply:
I'd love to send in a PR for that immediately after this!